### PR TITLE
Implement remove a selection from pre enrollment route

### PR DESCRIPTION
**Pull request by joseriveramorales**

### CODE REVIEW

File: premarum-backend/PreEnrollmentMgmt.Core/Services/StudentValidationService.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    private readonly IStudentRepository _studentRepository;
+    
+    public StudentValidationService(IStudentRepository studentRepository)
+    {
+        _studentRepository = studentRepository;
+    }
+    
+    public async Task<Boolean> ValidateStudentExists(string studentEmail)
+    {
+        var student = await _studentRepository.GetByEmailSimple(studentEmail);
```
[the remark this answers is not in the document]
Comment (joseriveramorales) [reply premarum-backend/PreEnrollmentMgmt.Core/Services/StudentValidationService.cs:17]: Sounds good!

File: premarum-backend/PreEnrollmentMgmt.Core/Services/PreEnrollmentService.cs [hunk 2]
```diff
@@ -49,6 +49,32 @@ public class PreEnrollmentService
...
+        if (preEnrollment.Selections.Count < courseOfferings.Length)
+            throw new InvalidPreEnrollmentSelectionException("Cannot remove more selections than what are available");
+
+        foreach (var semesterOffer in preEnrollment.Selections)
+        {
+            if(courseOfferings.Contains(semesterOffer.Id))
+                preEnrollment.RemoveSelection(semesterOffer);
+        }
+        _preEnrollmentRepository.Save(preEnrollment);
+        return "200 OK";
```
[the remark this answers is not in the document]
Comment (joseriveramorales) [reply premarum-backend/PreEnrollmentMgmt.Core/Services/PreEnrollmentService.cs:76]: Got it

File: premarum-backend/PreEnrollmentMgmt.Core/Services/PreEnrollmentService.cs [hunk 3]
```diff
@@ -49,6 +49,32 @@ public class PreEnrollmentService
...
+
+        var preEnrollment = await _preEnrollmentRepository.GetByIdWithSemesterOffersSimple(preEnrollmentId);
+
+        if (preEnrollment == null)
+            throw new InvalidPreEnrollmentSelectionException("PreEnrollment Not Found");
+        
+        if( ! await _studentValidationService.ValidateStudentCanModify(studentEmail, preEnrollment.StudentId))
+            throw new InvalidPreEnrollmentSelectionException("Student cannot change PreEnrollment");
+
+        if (preEnrollment.Selections.Count < courseOfferings.Length)
```
[the remark this answers is not in the document]
Comment (joseriveramorales) [reply premarum-backend/PreEnrollmentMgmt.Core/Services/PreEnrollmentService.cs:67]: I used the function initially, but Rider asked me to use the property, I honestly don't mind so I will use the function.

[the remark this answers is not in the document]
Comment (kenneth-rosario) [reply premarum-backend/PreEnrollmentMgmt.Core/Services/PreEnrollmentService.cs:67]: Oh maybe it has some optimization. Never mind then

File: premarum-backend/PreEnrollmentMgmt.Application/Repositories/PreEnrollmentRepository.cs [hunk 4]
```diff
@@ -27,6 +27,7 @@ public async Task<PreEnrollment> GetByIdWithSemesterOffersSimple(int preEnrollme
         return await _context
             .PreEnrollments
             .Include(pe => pe.Selections)
+            .Include(pe => pe.StudentId)
```
[the remark this answers is not in the document]
Comment (joseriveramorales) [reply premarum-backend/PreEnrollmentMgmt.Application/Repositories/PreEnrollmentRepository.cs:30]: Will remove this join.

**Pull request merged**